core/src/syscall/precompiles/fp384/fp12.rs

```diff
@@ -0,0 +1,1167 @@
+use crate::air::{MachineAir, Polynomial, SP1AirBuilder};
+use crate::bytes::event::ByteRecord;
+use crate::bytes::ByteLookupEvent;
+use crate::memory::{MemoryCols, MemoryReadCols, MemoryWriteCols};
+use crate::operations::field::field_op::{FieldOpCols, FieldOperation};
+use crate::operations::field::params::{FieldParameters, NumWords};
+use crate::operations::field::params::{Limbs, NumLimbs};
+use crate::runtime::{ExecutionRecord, Program, Syscall, SyscallCode};
+use crate::runtime::{MemoryReadRecord, MemoryWriteRecord};
+use crate::stark::MachineRecord;
+use crate::syscall::precompiles::SyscallContext;
+use crate::utils::ec::uint::U384Field;
+use crate::utils::{limbs_from_prev_access, pad_rows, words_to_bytes_le};
+use generic_array::GenericArray;
+use itertools::{izip, Itertools};
+use num::BigUint;
+use num::Zero;
+use p3_air::AirBuilder;
+use p3_air::{Air, BaseAir};
+use p3_field::AbstractField;
+use p3_field::PrimeField32;
+use p3_matrix::dense::RowMajorMatrix;
+use p3_matrix::Matrix;
+use serde::{Deserialize, Serialize};
+use sp1_derive::AlignedBorrow;
+use std::borrow::{Borrow, BorrowMut};
+use std::fmt::Debug;
+use std::marker::PhantomData;
+use std::mem::size_of;
+use typenum::Unsigned;
+
+use super::Fp12;
+
+/// The number of columns in the FpMulCols.
+const NUM_COLS: usize = size_of::<Fp12MulCols<u8>>();
+
+trait Fp12Builder<F> {
```

Comment:
The builder name is usually reserved for something that also satisfies `AirBuilder` in our crates.